### DISCUSSION TITLE
Better beginner examples?

### DISCUSSION THREAD

**Discussion by Sewdohe** [StarArawn/bevy_ecs_tilemap#539] [Q&A]
So I've tried rendering out a simple tilemap I've made but I'm having issues importing the "helper" files from the examples repo to get a simple render going.

When I add the "tiled.rs" file to my project it's just full of errors and I don't know how to solve them. I don't even know if it's an issue with my project or the helper files?

Can someone point me in the correct direction to render a simple tilemap?

**Comment by rparrett** [comment]
`tiled.rs` requires some modification when used outside of the `bevy_ecs_tilemap` repo.

The lines like:
```rust
 #[cfg(feature = "atlas")]
```
Are doing conditional compilation based on `bevy_ecs_tilemap`'s `atlas` feature, but in your own project, you won't necessarily have an `atlas` feature.

A very simple workaround is to add such a feature to your project with something like:
```toml
[features]
atlas = ["bevy_ecs_tilemap/atlas"]
```
In your `Cargo.toml`, which would simply enable the atlas feature for `bevy_ecs_tilemap` if the equivalent feature in your own project is enabled.

The `atlas` feature is a workaround for a lack of support for array textures in `webgl2` builds. If you know you will need it, or that you definitely won't need it, it's also possible to just remove the conditional compilation, leaving only the code paths you want.

There may also be various import paths that need adjusting if the structure of your project is different.

**Note**: the current disposition of the maintainer of this project is that the `tiled` and `ldtk` examples should be removed, as they are a maintenance burden. It's hoped for that a community project similar to [`bevy_ecs_ldtk`](https://github.com/Trouv/bevy_ecs_ldtk) will spring up for users who prefer the `Tiled` editor.

**Reply by Sewdohe** [reply]
Okay; that's more clear to me now. Thanks for the rundown.

I suppose this project has a bit of overhead for no more than I'm trying to do (just playing and learning atm) so I'll probably just switch to the ecs_ldtk plugin and the LDTK editor :)

I appreciate the explaination!
